country-dashboard/src/pages/Home.tsx

```diff
@@ -0,0 +1,205 @@
+          setLoader(false);
+        } else if (selectedLanguage !== "Any Language") {
+          setLoader(true);
+          url = `${baseUrl}/lang/${selectedLanguage}`;
+          const response = await fetch(url);
+          const data = await response.json();
+          setAllCountries(data);
+          setLoader(false);
+        } else if (selectedRegion !== "Any Region") {
+          setLoader(true);
+          url = `${baseUrl}/region/${selectedRegion}`;
+          const response = await fetch(url);
+          const data = await response.json();
+          setAllCountries(data);
+          setLoader(false);
+        }
+      } catch (error) {
+        console.error("Error fetching data:", error);
+      }
+    };
+
+    fetchData();
+  }, [searchValue, selectedCurrency, selectedLanguage, selectedRegion]);
+
+  // Event handlers
+  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setSearchValue(value);
+  };
+
+  const toggleDarkMode = () => {
+    setDarkMode(!darkMode);
+  };
+
+  const handleCurrencyChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const value = e.target.value;
+    setSelectedCurrency(value);
+  };
+
+  const handleLanguageChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const value = e.target.value;
+    setSelectedLanguage(value);
+  };
+
+  const handleRegionChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const value = e.target.value;
+    setSelectedRegion(value);
+  };
+
+  // JSX
+  return (
+    <div className={`container ${darkMode ? "dark-mode" : ""}`}>
+      <div>
+        <h3>Countries</h3>
+        <p>
+          <DarkModeSwitch
+            checked={darkMode}
+            onChange={toggleDarkMode}
+            size={30}
+          />
+        </p>
+      </div>
+      <header>
+        <div>
+          <SearchBar
+            id="searchBar"
+            type="text"
+            placeholder="Search..."
+            value={searchValue}
+            onChange={handleSearch}
+          />
+        </div>
+        <div>
+          <SelectOptions
+            onChange={handleCurrencyChange}
+            defaultValue={selectedCurrency}
+            name="currency"
+            id="currency"
+            options={currencyOptions}
+          />
+        </div>
+        <div>
+          <SelectOptions
+            onChange={handleLanguageChange}
+            defaultValue={selectedLanguage}
+            name="anyLanguage"
+            id="anyLanguage"
+            options={languageOptions}
+          />
+        </div>
+        <div>
+          <SelectOptions
+            onChange={handleRegionChange}
+            defaultValue={selectedRegion}
+            name="anyRegion"
+            id="anyRegion"
+            options={region.map((region) => region)}
+          />
+        </div>
+      </header>
+      <main>
+        {loader ? (
+          <img src={loadingPng} alt="Loading..." id="loader" />
```

Comment:
The loader is a bit disruptive. Make sure that your loader takes up the whole page, and then the loading image can be wherever. Otherwise, your app flickers when it loads, like it does now. -3